projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/search-ai-input/search-ai-input.component.spec.ts

```diff
@@ -396,6 +399,15 @@ describe('SearchAiInputComponent', () => {
 
         expect(dialogOpenSpy).toHaveBeenCalled();
       });
+
+      it('should open Unsaved Changes Modal', () => {
+        const modalAiSpy = spyOn(modalAi, 'openUnsavedChangesModal').and.callFake((callback) => callback());
```

Comment:
We should test what callback for openUnsavedChangesModal does. Actually we only check if that callback was called but we don't know what happens after that so we don't know if functionality works after calling callback.

Reply:
I see my comment confused you a bit because I meant following: 
<img width="962" alt="image" src="https://github.com/user-attachments/assets/1de276a0-1774-48bf-9ba8-e6f120607b3a">

You can see that as callback we call search function so I meant that we can't test it in this way, we need to ensure that as callback logic from that function is triggered. In other case if we pass mocked function as callback then we still don't know if that logic works.